scripts/template_matcher.py

```diff
@@ -62,12 +63,73 @@ def _compute_prediction(self, k, img, kp, des):
         # TODO: find corresponding points in the input image and the templae image
         #       put keypoints from template image in template_pts
         #       put corresponding keypoints from input image in img_pts
-
-
-        #TODO: change img to img_T once you do the homography transform
-        visual_diff = compare_images(img, self.signs[k])
-        return visual_diff
+        img_kp, img_des = self.orb.detectAndCompute(img,None)
+
+        template_pts = np.array([[0.,0.]])
+        img_pts = np.array([[0.,0.]])
+        
+        for i in xrange(len(des[k])):
+            descriptor = des[k][i]
+            closest = [False, False]
+            second_closest = [False, False]
+            for j in xrange(len(img_des)):
+                img_descriptor = img_des[j]
+                dist = np.linalg.norm(descriptor - img_descriptor, 2)
+                if closest[0]:
+                    if dist < closest[1]:
+                        second_closest[0] = closest[0]
+                        second_closest[1] = closest[1]
+                        closest[0] = j
+                        closest[1] = dist
+                    elif (not second_closest[0]) or (dist < second_closest[1]):
+                        second_closest[0] = j
+                        second_closest[1] = dist
+                else:
+                    closest[0] = j
+                    closest[1] = dist
+            if closest[1] < (0.7*second_closest[1]):
+                template_pts = np.append(template_pts, [[kp[k][i].pt[0], kp[k][i].pt[1]]], axis=0)
+                img_pts = np.append(img_pts, [[img_kp[i].pt[0], img_kp[i].pt[1]]], axis=0)
+
+        if len(img_pts) > 2:
```

Comment:
I don't understand this one fully, why image_pts should be > 2.  I *thought* that image_pts should be >= 4 since you need 4 points to define a homography.